### PR TITLE
Push web3-eth Test Coverage above 80%

### LINKED ISSUE #5930: Coverage Web3 Eth
push unit tests coverage >= 80% of web3 eth https://app.codecov.io/gh/web3/web3.js/tree/4.x/packages

### PR DESCRIPTION
**Pull request by Muhammad-Altabba**
## Description

Fixes #5930

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist for 1.x:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` with success.
- [ ] I have tested the built `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.

## Checklist for 4.x:

- [ ] I have selected the correct 4.x base branch.
- [ ] Within the description, the feature or issue is discussed in detail or an issue is linked.   
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added any required tests for the changes I made 
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `yarn` successfully
- [ ] I ran `yarn lint` successfully
- [ ] I ran `yarn build:web` successfully
- [ ] I ran `yarn test:unit` successfully
- [ ] I ran `yarn test:integration` successfully
- [ ] I ran `compile:contracts` successfully
- [ ] I have tested my code.
- [ ] I have updated the corresponding `CHANGELOG.md` file in the packages I have edited.

### CODE REVIEW

File: packages/web3-eth/src/web3_eth.ts [hunk 1]
```diff
@@ -1568,7 +1568,7 @@ export class Web3Eth extends Web3Context<Web3EthExecutionAPI, RegisteredSubscrip
 			subscription instanceof LogsSubscription &&
 			name === 'logs' &&
 			typeof args === 'object' &&
-			args.fromBlock &&
+			(args.fromBlock || args.fromBlock === 0) &&
```
Comment (Spacesai1or) [inline packages/web3-eth/src/web3_eth.ts]: I think we should replace this with `!isNullish(args.fromBlock)`

File: packages/web3-eth/src/utils/watch_transaction_by_subscription.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+ */
+export const watchTransactionBySubscription = <
+	ReturnFormat extends DataFormat,
+	ResolveType = TransactionReceipt,
+>({
+	web3Context,
+	transactionReceipt,
+	transactionPromiEvent,
+	returnFormat,
+}: WaitProps<ReturnFormat, ResolveType>) => {
```
Comment (jdevcs) [inline packages/web3-eth/src/utils/watch_transaction_by_subscription.ts:36]: some of these lines are not covered so patch is 50% in code cov. If its moved functionality wasn't it covered in past?

**Review by jdevcs — APPROVED** [review]

File: packages/web3-eth/src/utils/watch_transaction_by_subscription.ts (hunk 2 above)
Comment (Muhammad-Altabba) [reply packages/web3-eth/src/utils/watch_transaction_by_subscription.ts:36]: Yes, it is just moved from another file. I separate it to ease the testing of other functions. And it was not covered in the past in the unit tests. Actually, it is only, but well-tested, in the integration test.

File: packages/web3-eth/src/web3_eth.ts (hunk 1 above)
Comment (Muhammad-Altabba) [reply packages/web3-eth/src/web3_eth.ts]: done

**Review by nikoulai — APPROVED** [review]

**Pull request merged**